### ISSUE TITLE
JSDoc 3.4.3 does nothing

### ISSUE THREAD

**Issue by ZelphirKaltstahl** [jsdoc3/jsdoc#1372]
When I run ...

    jsdoc --recurse --verbose --destination docs my-folder

... according to the [Getting Startet](http://usejsdoc.org/about-getting-started.html) guide, the result is only a HTML page with no documentation whatsoever on it.

My source code is an AMD and has nested functions (I think down to level 3 nesting), some of them commented according to the getting started guide.

What is wrong with my jsdoc?

# Info

* jsdoc version: `JSDoc 3.4.3 (Thu, 10 Nov 2016 00:25:10 GMT)`
* installation: `npm install -g jsdoc`
* `npm --version`: `5.0.0`
* `node --version`:  `v8.0.0`

**Comment by rsxdalv** [comment]
Same here

**Comment by iPherian** [comment]
I found that I needed to add at least one `@namespace` statement before anything appeared.

See http://usejsdoc.org/tags-namespace.html

**Comment by hegemonic** [comment]
I can't answer this question without an example of the code that's not generating documentation. If you can provide an example, I'll reopen this issue.

**Comment by ZelphirKaltstahl** [comment]
A Jupyter Notebook extension:

    // This is AMD notation.
    define(
        [
            'jquery',
            'underscore',
            'bootstrap',
            'base/js/dialog',
            'base/js/namespace',
            'base/js/events',
            'require'
        ],
        function($, _, bootstrap, dialog, Jupyter, events, require) {
            // inside here only functions and functions in functions
        }
    );

**Comment by hegemonic** [comment]
@ZelphirKaltstahl, you have to add JSDoc comments in order for JSDoc to generate documentation. See the [instructions for documenting AMD modules](http://usejsdoc.org/howto-amd-modules.html) for examples.

**Comment by ZelphirKaltstahl** [comment]
Ah I see. So there is something special for AMDs, I did not know that.
I think it's not so practical, that the fact, that AMDs need different treatment or that there are structures which need different treatment at all is not mentioned in [getting-started](http://usejsdoc.org/about-getting-started.html). Many people might go there try it, see nothing works and turn their backs.

In the meantime I already wrote my documentation in a separate document myself. Maybe in the future I will use JSDoc.
